test/datacollector.UnitTests/DataCollectionAttachmentManagerTests.cs

```diff
@@ -39,6 +44,47 @@ public void Cleanup()
         File.Delete(Path.Combine(TempDirectoryPath, "filename1.txt"));
     }
 
+    [TestMethod]
+    public void ParallelAccessShouldNotBreak()
```

Comment:
That's really cool you found a way to replicate the issue <3

Reply:
Rare but it happen :(